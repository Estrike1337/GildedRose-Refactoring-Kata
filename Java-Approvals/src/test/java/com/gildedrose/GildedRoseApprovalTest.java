package com.gildedrose;

import org.approvaltests.Approvals;
import org.approvaltests.combinations.CombinationApprovals;
import org.approvaltests.reporters.DiffReporter;
import org.approvaltests.reporters.UseReporter;
import org.junit.jupiter.api.Test;

import java.io.ByteArrayInputStream;
import java.io.ByteArrayOutputStream;
import java.io.PrintStream;

@UseReporter(DiffReporter.class)
public class GildedRoseApprovalTest {

    @Test
    public void foo() {

        //        Item[] items = new Item[]{new Item("+5 Dexterity Vest", 10, 20), //
        //            new Item("Aged Brie", 2, 0), //
        //            new Item("Elixir of the Mongoose", 5, 7), //
        //            new Item("Sulfuras, Hand of Ragnaros", 0, 80), //
        //            new Item("Sulfuras, Hand of Ragnaros", -1, 80),
        //            new Item("Backstage passes to a TAFKAL80ETC concert", 15, 20),
        //            new Item("Backstage passes to a TAFKAL80ETC concert", 10, 49),
        //            new Item("Backstage passes to a TAFKAL80ETC concert", 5, 49),
        //            // this conjured item does not work properly yet
        //            new Item("Conjured Mana Cake", 3, 6)};
        //        GildedRose app = new GildedRose(items);
        //        app.updateQuality();
        String[] names =
            new String[]{"+5 Dexterity Vest", "Aged Brie", "Elixir of the Mongoose", "Sulfuras, Hand of Ragnaros",
                "Sulfuras, Hand of Ragnaros", "Backstage passes to a TAFKAL80ETC concert",
                "Backstage passes to a TAFKAL80ETC concert", "Backstage passes to a TAFKAL80ETC concert",
                "Conjured Mana Cake", "foo"};
        Integer[] sellIns = new Integer[]{0, 12, 5, -1};
        Integer[] qualities = new Integer[]{0, 1, 5, 50, 49};
        CombinationApprovals.verifyAllCombinations(this::doUpdateQuality, names, sellIns, qualities);
        //Approvals.verifyAll("Items", items);
    }

    public String doUpdateQuality(String name, int sellIn, int quality) {
        Item[] items = new Item[]{new Item(name, sellIn, quality)};
        GildedRose app = new GildedRose(items);
        app.updateQuality();
        return app.items[0].toString();
    }
}
