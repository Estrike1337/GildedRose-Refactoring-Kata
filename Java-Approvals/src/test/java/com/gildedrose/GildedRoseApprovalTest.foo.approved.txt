[+5 Dexterity Vest, 0, 0] => +5 Dexterity Vest, 0, 0 
[+5 Dexterity Vest, 0, 1] => +5 Dexterity Vest, 0, 1 
[+5 Dexterity Vest, 0, 5] => +5 Dexterity Vest, 0, 5 
[+5 Dexterity Vest, 0, 50] => +5 Dexterity Vest, 0, 50 
[+5 Dexterity Vest, 0, 49] => +5 Dexterity Vest, 0, 49 
[+5 Dexterity Vest, 12, 0] => +5 Dexterity Vest, 12, 0 
[+5 Dexterity Vest, 12, 1] => +5 Dexterity Vest, 12, 1 
[+5 Dexterity Vest, 12, 5] => +5 Dexterity Vest, 12, 5 
[+5 Dexterity Vest, 12, 50] => +5 Dexterity Vest, 12, 50 
[+5 Dexterity Vest, 12, 49] => +5 Dexterity Vest, 12, 49 
[+5 Dexterity Vest, 5, 0] => +5 Dexterity Vest, 5, 0 
[+5 Dexterity Vest, 5, 1] => +5 Dexterity Vest, 5, 1 
[+5 Dexterity Vest, 5, 5] => +5 Dexterity Vest, 5, 5 
[+5 Dexterity Vest, 5, 50] => +5 Dexterity Vest, 5, 50 
[+5 Dexterity Vest, 5, 49] => +5 Dexterity Vest, 5, 49 
[+5 Dexterity Vest, -1, 0] => +5 Dexterity Vest, -1, 0 
[+5 Dexterity Vest, -1, 1] => +5 Dexterity Vest, -1, 1 
[+5 Dexterity Vest, -1, 5] => +5 Dexterity Vest, -1, 5 
[+5 Dexterity Vest, -1, 50] => +5 Dexterity Vest, -1, 50 
[+5 Dexterity Vest, -1, 49] => +5 Dexterity Vest, -1, 49 
[Aged Brie, 0, 0] => Aged Brie, -1, 2 
[Aged Brie, 0, 1] => Aged Brie, -1, 3 
[Aged Brie, 0, 5] => Aged Brie, -1, 7 
[Aged Brie, 0, 50] => Aged Brie, -1, 50 
[Aged Brie, 0, 49] => Aged Brie, -1, 50 
[Aged Brie, 12, 0] => Aged Brie, 11, 1 
[Aged Brie, 12, 1] => Aged Brie, 11, 2 
[Aged Brie, 12, 5] => Aged Brie, 11, 6 
[Aged Brie, 12, 50] => Aged Brie, 11, 50 
[Aged Brie, 12, 49] => Aged Brie, 11, 50 
[Aged Brie, 5, 0] => Aged Brie, 4, 1 
[Aged Brie, 5, 1] => Aged Brie, 4, 2 
[Aged Brie, 5, 5] => Aged Brie, 4, 6 
[Aged Brie, 5, 50] => Aged Brie, 4, 50 
[Aged Brie, 5, 49] => Aged Brie, 4, 50 
[Aged Brie, -1, 0] => Aged Brie, -2, 2 
[Aged Brie, -1, 1] => Aged Brie, -2, 3 
[Aged Brie, -1, 5] => Aged Brie, -2, 7 
[Aged Brie, -1, 50] => Aged Brie, -2, 50 
[Aged Brie, -1, 49] => Aged Brie, -2, 50 
[Elixir of the Mongoose, 0, 0] => Elixir of the Mongoose, 0, 0 
[Elixir of the Mongoose, 0, 1] => Elixir of the Mongoose, 0, 1 
[Elixir of the Mongoose, 0, 5] => Elixir of the Mongoose, 0, 5 
[Elixir of the Mongoose, 0, 50] => Elixir of the Mongoose, 0, 50 
[Elixir of the Mongoose, 0, 49] => Elixir of the Mongoose, 0, 49 
[Elixir of the Mongoose, 12, 0] => Elixir of the Mongoose, 12, 0 
[Elixir of the Mongoose, 12, 1] => Elixir of the Mongoose, 12, 1 
[Elixir of the Mongoose, 12, 5] => Elixir of the Mongoose, 12, 5 
[Elixir of the Mongoose, 12, 50] => Elixir of the Mongoose, 12, 50 
[Elixir of the Mongoose, 12, 49] => Elixir of the Mongoose, 12, 49 
[Elixir of the Mongoose, 5, 0] => Elixir of the Mongoose, 5, 0 
[Elixir of the Mongoose, 5, 1] => Elixir of the Mongoose, 5, 1 
[Elixir of the Mongoose, 5, 5] => Elixir of the Mongoose, 5, 5 
[Elixir of the Mongoose, 5, 50] => Elixir of the Mongoose, 5, 50 
[Elixir of the Mongoose, 5, 49] => Elixir of the Mongoose, 5, 49 
[Elixir of the Mongoose, -1, 0] => Elixir of the Mongoose, -1, 0 
[Elixir of the Mongoose, -1, 1] => Elixir of the Mongoose, -1, 1 
[Elixir of the Mongoose, -1, 5] => Elixir of the Mongoose, -1, 5 
[Elixir of the Mongoose, -1, 50] => Elixir of the Mongoose, -1, 50 
[Elixir of the Mongoose, -1, 49] => Elixir of the Mongoose, -1, 49 
[Sulfuras, Hand of Ragnaros, 0, 0] => Sulfuras, Hand of Ragnaros, 0, 0 
[Sulfuras, Hand of Ragnaros, 0, 1] => Sulfuras, Hand of Ragnaros, 0, 1 
[Sulfuras, Hand of Ragnaros, 0, 5] => Sulfuras, Hand of Ragnaros, 0, 5 
[Sulfuras, Hand of Ragnaros, 0, 50] => Sulfuras, Hand of Ragnaros, 0, 50 
[Sulfuras, Hand of Ragnaros, 0, 49] => Sulfuras, Hand of Ragnaros, 0, 49 
[Sulfuras, Hand of Ragnaros, 12, 0] => Sulfuras, Hand of Ragnaros, 12, 0 
[Sulfuras, Hand of Ragnaros, 12, 1] => Sulfuras, Hand of Ragnaros, 12, 1 
[Sulfuras, Hand of Ragnaros, 12, 5] => Sulfuras, Hand of Ragnaros, 12, 5 
[Sulfuras, Hand of Ragnaros, 12, 50] => Sulfuras, Hand of Ragnaros, 12, 50 
[Sulfuras, Hand of Ragnaros, 12, 49] => Sulfuras, Hand of Ragnaros, 12, 49 
[Sulfuras, Hand of Ragnaros, 5, 0] => Sulfuras, Hand of Ragnaros, 5, 0 
[Sulfuras, Hand of Ragnaros, 5, 1] => Sulfuras, Hand of Ragnaros, 5, 1 
[Sulfuras, Hand of Ragnaros, 5, 5] => Sulfuras, Hand of Ragnaros, 5, 5 
[Sulfuras, Hand of Ragnaros, 5, 50] => Sulfuras, Hand of Ragnaros, 5, 50 
[Sulfuras, Hand of Ragnaros, 5, 49] => Sulfuras, Hand of Ragnaros, 5, 49 
[Sulfuras, Hand of Ragnaros, -1, 0] => Sulfuras, Hand of Ragnaros, -1, 0 
[Sulfuras, Hand of Ragnaros, -1, 1] => Sulfuras, Hand of Ragnaros, -1, 1 
[Sulfuras, Hand of Ragnaros, -1, 5] => Sulfuras, Hand of Ragnaros, -1, 5 
[Sulfuras, Hand of Ragnaros, -1, 50] => Sulfuras, Hand of Ragnaros, -1, 50 
[Sulfuras, Hand of Ragnaros, -1, 49] => Sulfuras, Hand of Ragnaros, -1, 49 
[Sulfuras, Hand of Ragnaros, 0, 0] => Sulfuras, Hand of Ragnaros, 0, 0 
[Sulfuras, Hand of Ragnaros, 0, 1] => Sulfuras, Hand of Ragnaros, 0, 1 
[Sulfuras, Hand of Ragnaros, 0, 5] => Sulfuras, Hand of Ragnaros, 0, 5 
[Sulfuras, Hand of Ragnaros, 0, 50] => Sulfuras, Hand of Ragnaros, 0, 50 
[Sulfuras, Hand of Ragnaros, 0, 49] => Sulfuras, Hand of Ragnaros, 0, 49 
[Sulfuras, Hand of Ragnaros, 12, 0] => Sulfuras, Hand of Ragnaros, 12, 0 
[Sulfuras, Hand of Ragnaros, 12, 1] => Sulfuras, Hand of Ragnaros, 12, 1 
[Sulfuras, Hand of Ragnaros, 12, 5] => Sulfuras, Hand of Ragnaros, 12, 5 
[Sulfuras, Hand of Ragnaros, 12, 50] => Sulfuras, Hand of Ragnaros, 12, 50 
[Sulfuras, Hand of Ragnaros, 12, 49] => Sulfuras, Hand of Ragnaros, 12, 49 
[Sulfuras, Hand of Ragnaros, 5, 0] => Sulfuras, Hand of Ragnaros, 5, 0 
[Sulfuras, Hand of Ragnaros, 5, 1] => Sulfuras, Hand of Ragnaros, 5, 1 
[Sulfuras, Hand of Ragnaros, 5, 5] => Sulfuras, Hand of Ragnaros, 5, 5 
[Sulfuras, Hand of Ragnaros, 5, 50] => Sulfuras, Hand of Ragnaros, 5, 50 
[Sulfuras, Hand of Ragnaros, 5, 49] => Sulfuras, Hand of Ragnaros, 5, 49 
[Sulfuras, Hand of Ragnaros, -1, 0] => Sulfuras, Hand of Ragnaros, -1, 0 
[Sulfuras, Hand of Ragnaros, -1, 1] => Sulfuras, Hand of Ragnaros, -1, 1 
[Sulfuras, Hand of Ragnaros, -1, 5] => Sulfuras, Hand of Ragnaros, -1, 5 
[Sulfuras, Hand of Ragnaros, -1, 50] => Sulfuras, Hand of Ragnaros, -1, 50 
[Sulfuras, Hand of Ragnaros, -1, 49] => Sulfuras, Hand of Ragnaros, -1, 49 
[Backstage passes to a TAFKAL80ETC concert, 0, 0] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 1] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 5] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 50] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 49] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 12, 0] => Backstage passes to a TAFKAL80ETC concert, 11, 1 
[Backstage passes to a TAFKAL80ETC concert, 12, 1] => Backstage passes to a TAFKAL80ETC concert, 11, 2 
[Backstage passes to a TAFKAL80ETC concert, 12, 5] => Backstage passes to a TAFKAL80ETC concert, 11, 6 
[Backstage passes to a TAFKAL80ETC concert, 12, 50] => Backstage passes to a TAFKAL80ETC concert, 11, 50 
[Backstage passes to a TAFKAL80ETC concert, 12, 49] => Backstage passes to a TAFKAL80ETC concert, 11, 50 
[Backstage passes to a TAFKAL80ETC concert, 5, 0] => Backstage passes to a TAFKAL80ETC concert, 4, 3 
[Backstage passes to a TAFKAL80ETC concert, 5, 1] => Backstage passes to a TAFKAL80ETC concert, 4, 4 
[Backstage passes to a TAFKAL80ETC concert, 5, 5] => Backstage passes to a TAFKAL80ETC concert, 4, 8 
[Backstage passes to a TAFKAL80ETC concert, 5, 50] => Backstage passes to a TAFKAL80ETC concert, 4, 50 
[Backstage passes to a TAFKAL80ETC concert, 5, 49] => Backstage passes to a TAFKAL80ETC concert, 4, 50 
[Backstage passes to a TAFKAL80ETC concert, -1, 0] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 1] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 5] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 50] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 49] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 0] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 1] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 5] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 50] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 49] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 12, 0] => Backstage passes to a TAFKAL80ETC concert, 11, 1 
[Backstage passes to a TAFKAL80ETC concert, 12, 1] => Backstage passes to a TAFKAL80ETC concert, 11, 2 
[Backstage passes to a TAFKAL80ETC concert, 12, 5] => Backstage passes to a TAFKAL80ETC concert, 11, 6 
[Backstage passes to a TAFKAL80ETC concert, 12, 50] => Backstage passes to a TAFKAL80ETC concert, 11, 50 
[Backstage passes to a TAFKAL80ETC concert, 12, 49] => Backstage passes to a TAFKAL80ETC concert, 11, 50 
[Backstage passes to a TAFKAL80ETC concert, 5, 0] => Backstage passes to a TAFKAL80ETC concert, 4, 3 
[Backstage passes to a TAFKAL80ETC concert, 5, 1] => Backstage passes to a TAFKAL80ETC concert, 4, 4 
[Backstage passes to a TAFKAL80ETC concert, 5, 5] => Backstage passes to a TAFKAL80ETC concert, 4, 8 
[Backstage passes to a TAFKAL80ETC concert, 5, 50] => Backstage passes to a TAFKAL80ETC concert, 4, 50 
[Backstage passes to a TAFKAL80ETC concert, 5, 49] => Backstage passes to a TAFKAL80ETC concert, 4, 50 
[Backstage passes to a TAFKAL80ETC concert, -1, 0] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 1] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 5] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 50] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 49] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 0] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 1] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 5] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 50] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 49] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 12, 0] => Backstage passes to a TAFKAL80ETC concert, 11, 1 
[Backstage passes to a TAFKAL80ETC concert, 12, 1] => Backstage passes to a TAFKAL80ETC concert, 11, 2 
[Backstage passes to a TAFKAL80ETC concert, 12, 5] => Backstage passes to a TAFKAL80ETC concert, 11, 6 
[Backstage passes to a TAFKAL80ETC concert, 12, 50] => Backstage passes to a TAFKAL80ETC concert, 11, 50 
[Backstage passes to a TAFKAL80ETC concert, 12, 49] => Backstage passes to a TAFKAL80ETC concert, 11, 50 
[Backstage passes to a TAFKAL80ETC concert, 5, 0] => Backstage passes to a TAFKAL80ETC concert, 4, 3 
[Backstage passes to a TAFKAL80ETC concert, 5, 1] => Backstage passes to a TAFKAL80ETC concert, 4, 4 
[Backstage passes to a TAFKAL80ETC concert, 5, 5] => Backstage passes to a TAFKAL80ETC concert, 4, 8 
[Backstage passes to a TAFKAL80ETC concert, 5, 50] => Backstage passes to a TAFKAL80ETC concert, 4, 50 
[Backstage passes to a TAFKAL80ETC concert, 5, 49] => Backstage passes to a TAFKAL80ETC concert, 4, 50 
[Backstage passes to a TAFKAL80ETC concert, -1, 0] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 1] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 5] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 50] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 49] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Conjured Mana Cake, 0, 0] => Conjured Mana Cake, -1, 0 
[Conjured Mana Cake, 0, 1] => Conjured Mana Cake, -1, 0 
[Conjured Mana Cake, 0, 5] => Conjured Mana Cake, -1, 1 
[Conjured Mana Cake, 0, 50] => Conjured Mana Cake, -1, 46 
[Conjured Mana Cake, 0, 49] => Conjured Mana Cake, -1, 45 
[Conjured Mana Cake, 12, 0] => Conjured Mana Cake, 11, 0 
[Conjured Mana Cake, 12, 1] => Conjured Mana Cake, 11, 0 
[Conjured Mana Cake, 12, 5] => Conjured Mana Cake, 11, 3 
[Conjured Mana Cake, 12, 50] => Conjured Mana Cake, 11, 48 
[Conjured Mana Cake, 12, 49] => Conjured Mana Cake, 11, 47 
[Conjured Mana Cake, 5, 0] => Conjured Mana Cake, 4, 0 
[Conjured Mana Cake, 5, 1] => Conjured Mana Cake, 4, 0 
[Conjured Mana Cake, 5, 5] => Conjured Mana Cake, 4, 3 
[Conjured Mana Cake, 5, 50] => Conjured Mana Cake, 4, 48 
[Conjured Mana Cake, 5, 49] => Conjured Mana Cake, 4, 47 
[Conjured Mana Cake, -1, 0] => Conjured Mana Cake, -2, 0 
[Conjured Mana Cake, -1, 1] => Conjured Mana Cake, -2, 0 
[Conjured Mana Cake, -1, 5] => Conjured Mana Cake, -2, 1 
[Conjured Mana Cake, -1, 50] => Conjured Mana Cake, -2, 46 
[Conjured Mana Cake, -1, 49] => Conjured Mana Cake, -2, 45 
[foo, 0, 0] => foo, 0, 0 
[foo, 0, 1] => foo, 0, 1 
[foo, 0, 5] => foo, 0, 5 
[foo, 0, 50] => foo, 0, 50 
[foo, 0, 49] => foo, 0, 49 
[foo, 12, 0] => foo, 12, 0 
[foo, 12, 1] => foo, 12, 1 
[foo, 12, 5] => foo, 12, 5 
[foo, 12, 50] => foo, 12, 50 
[foo, 12, 49] => foo, 12, 49 
[foo, 5, 0] => foo, 5, 0 
[foo, 5, 1] => foo, 5, 1 
[foo, 5, 5] => foo, 5, 5 
[foo, 5, 50] => foo, 5, 50 
[foo, 5, 49] => foo, 5, 49 
[foo, -1, 0] => foo, -1, 0 
[foo, -1, 1] => foo, -1, 1 
[foo, -1, 5] => foo, -1, 5 
[foo, -1, 50] => foo, -1, 50 
[foo, -1, 49] => foo, -1, 49 
